controllers/profiles/dev/states_dev.go

```diff
@@ -62,23 +62,26 @@ func (e *ensureRunningWorkflowState) CanReconcile(workflow *operatorapi.SonataFl
 func (e *ensureRunningWorkflowState) Do(ctx context.Context, workflow *operatorapi.SonataFlow) (ctrl.Result, []client.Object, error) {
 	var objs []client.Object
 
-	flowDefCM, _, err := e.ensurers.definitionConfigMap.Ensure(ctx, workflow, ensureWorkflowDefConfigMapMutator(workflow))
-	if err != nil {
-		return ctrl.Result{Requeue: false}, objs, err
-	}
-	objs = append(objs, flowDefCM)
-
 	devBaseContainerImage := workflowdef.GetDefaultWorkflowDevModeImageTag()
 	// check if the Platform available
 	pl, err := platform.GetActivePlatform(ctx, e.C, workflow.Namespace)
 	if err == nil && len(pl.Spec.DevMode.BaseImage) > 0 {
 		devBaseContainerImage = pl.Spec.DevMode.BaseImage
 	}
-	propsCM, _, err := e.ensurers.propertiesConfigMap.Ensure(ctx, workflow, common.WorkflowPropertiesMutateVisitor(ctx, e.StateSupport.Catalog, workflow, pl))
+	flowDefCM, _, err := e.ensurers.definitionConfigMap.Ensure(ctx, workflow, pl, ensureWorkflowDefConfigMapMutator(workflow, pl))
+	if err != nil {
+		return ctrl.Result{Requeue: false}, objs, err
+	}
+	objs = append(objs, flowDefCM)
+	userPropsCM, _, err := e.ensurers.userPropsConfigMap.Ensure(ctx, workflow, pl)
+	if err != nil {
+		return ctrl.Result{Requeue: false}, objs, err
+	}
+	managedPropsCM, _, err := e.ensurers.managedPropsConfigMap.Ensure(ctx, workflow, pl, common.UserPropertiesMutateVisitor(ctx, e.StateSupport.Catalog, workflow, pl, userPropsCM.(*corev1.ConfigMap)))
```

Comment:
Does the `managedProps` include user properties too? Is this to calculate hash as you are describing in the PR? Can't we let the managed CM hold only manage props?

Reply:
Yes, the idea is to hold only managed props, but the checksum is calculated on the aggregated string "user props, managed props", in order to detect any possible change

Reply:
I see, but maybe we can do in a different function? Not sure if I followed the code exactly but it seems confusing having user and managed properties being merged in the function responsible by creating the CM. Maybe a `calculateHash(user, managed)` function?

Reply:
please verify the latest commit.